cdap-watchdog/src/main/java/io/cdap/cdap/logging/guice/LocalLogAppenderModule.java

```diff
@@ -28,6 +34,16 @@ public class LocalLogAppenderModule extends AbstractModule {
 
   @Override
   protected void configure() {
-    bind(LogAppender.class).to(LocalLogAppender.class).in(Scopes.SINGLETON);
+    bind(LocalLogAppender.class).in(Scopes.SINGLETON);
+    bind(ExtensionLogAppender.class).in(Scopes.SINGLETON);
+  }
+
+  @Provides
+  @Singleton
+  @SuppressWarnings("unused")
+  protected LogAppender provideCompositeLogAppender(LocalLogAppender localLogAppender,
+      ExtensionLogAppender extensionLogAppender) {
+    List<LogAppender> appenders = Arrays.asList(localLogAppender, extensionLogAppender);
```

Comment:
I would consider checking if publisher is enabled here and skip creating Composite if it's not.